### ISSUE TITLE
If I touch and hold toast, it never hides

### ISSUE THREAD

**Issue by calintamas** [calintamas/react-native-toast-message#299]
Not sure if it is intentional behavior, but if I touch and hold toast, it never hides (`autoHide` is set to `true`). The hiding animation takes place, but toast stays on screen infinitely. You may need to pull your finger down a bit after the hiding animation is complete in order to display toast again. I get this behavior on iOS as well as Android. Package version: 2.1.0

_Originally posted by @danieldaukapp in https://github.com/calintamas/react-native-toast-message/issues/280#issuecomment-990677323_

**Comment by babar7** [comment]
Yeah, also facing the same issue, touch and swipe lil bit down and hold then release, it never hides. And if you hold longer it hide and reappear  when release the touch.

**Comment by pankaj-k-simformsolutions** [comment]
@calintamas  any update on this

**Comment by doougui** [comment]
Facing the same issue here.

**Comment by sushantWemotive** [comment]
same issue here

**Comment by mrshahzeb7** [comment]
any solution ?

**Comment by mrshahzeb7** [comment]
used this logic 
it is working for me 

  const [isVisibleToast, setisVisibleToast] = useState(false);

  useEffect(() => {
    if (isVisible) {
      setisVisibleToast(true);
    } else {
      setTimeout(() => {
        setisVisibleToast(false);
      }, 2000);
    }
  }, [isVisible]);

  return isVisibleToast ? (
    <View style={styles.container}>
      {
        //custom Toast
      }
      </View>)
    :
    null

**Comment by oakmegaeddie** [comment]
I guess the reason is that after `visibilityTime`, the toast is still being pressed, so it can't hide after release.
The solution to fix this bug in the library code would be check whether it should hide or not after release.

**Comment by KoenVerhulst** [comment]
Any plans in fixing this?

**Comment by knightmate** [comment]
@calintamas Hi, do you have any updates on this? I've also encountered this bug on Android. Should I address it?

**Comment by mleister97** [comment]
The issue persists in 2024...

**Comment by pepperinpeppermint314159** [comment]
node_modules/react-native-toast-message/lib/src/components/AnimatedContainer.d.ts

change this:
`const { dy, numberActiveTouches } = gesture; // use numberActiveTouches
    switch (position) {
        case 'bottom':
            return boundValue(1 - dy / damping);
        case 'top':
            if (!numberActiveTouches) return boundValue(0) // check if numberActiveTouches == 0 
            return boundValue(1 + dy / damping);
        default:
            throw new Error(`Toast position: ${position} not implemented`);
}`

**Comment by Meet-Kapadiya** [comment]
Still it recreates a new instance

**Comment by Stelchenko** [comment]
@Meet-Kapadiya Can you tell me how you made the timer? In my implementation it has a bug when the toast disappears. (The timer returns to the original position 0% )

**Comment by athaljen** [comment]
> @Meet-Kapadiya Can you tell me how you made the timer? In my implementation it has a bug when the toast disappears. (The timer returns to the original position 0% )

```js

const fillAnimation = React.createRef<Animated.Value>();

const closeHandler = () => {
  Toast.hide();
};

const toastConfig: ToastConfig = {
  success: (params: ToastConfigParams<ToastProps>) => {
    const {text, visibilityTime} = params.props;

    // @ts-ignore
    fillAnimation.current = new Animated.Value(0);

    Animated.timing(fillAnimation.current, {
      toValue: 1,
      duration: visibilityTime,
      useNativeDriver: false,
    }).start();

    return (
      <View style={styles.toast}>
        <View style={styles.toastMsgCont}>
          <Pressable
            style={styles.toastIconCont}
            onPress={closeHandler}
            hitSlop={8}>
            <FastImage
              source={Images.success}
              resizeMode="contain"
              style={styles.toastIcon}
            />
          </Pressable>
          <Text style={styles.toastText} numberOfLines={4}>
            {text}
          </Text>
        </View>
        <Animated.View
          style={[
            styles.toastIndicator,
            {
              width: fillAnimation.current.interpolate({
                inputRange: [0, 1],
                outputRange: ['0%', '100%'],
              }),
              backgroundColor: Colors.greenF80,
            },
          ]}
        />
      </View>
    );
  },
  error: (params: ToastConfigParams<ToastProps>) => {
    const {text, visibilityTime} = params.props;

    // @ts-ignore
    fillAnimation.current = new Animated.Value(0);

    Animated.timing(fillAnimation.current, {
      toValue: 1,
      duration: visibilityTime,
      useNativeDriver: false,
    }).start();

    return (
      <View style={styles.toast}>
        <View style={styles.toastMsgCont}>
          <Pressable
            style={styles.toastIconCont}
            onPress={closeHandler}
            hitSlop={8}>
            <FastImage
              source={Images.error}
              resizeMode="contain"
              style={styles.toastIcon}
            />
          </Pressable>
          <Text style={styles.toastText} numberOfLines={4}>
            {text}
          </Text>
        </View>
        <Animated.View
          style={[
            styles.toastIndicator,
            {
              width: fillAnimation.current.interpolate({
                inputRange: [0, 1],
                outputRange: ['0%', '100%'],
              }),
              backgroundColor: Colors.redD65,
            },
          ]}
        />
      </View>
    );
  },
};
```

**Comment by nayan-dey** [comment]
> node_modules/react-native-toast-message/lib/src/components/AnimatedContainer.d.ts in function: animatedValueFor
> 
> add this:
> 
> ```
>     const { dy, numberActiveTouches } = gesture;  // use numberActiveTouches
> 
>     switch (position) {
>         case 'bottom':
>             if (!numberActiveTouches) return boundValue(0) // add this
>             return boundValue(1 - dy / damping);
>         case 'top':
>             if (!numberActiveTouches) return boundValue(0) // add this
>             return boundValue(1 + dy / damping);
>         default:
>             throw new Error(`Toast position: ${position} not implemented`);
>     }
> ```

Thanks mate it solves the issue 

quick correction the function will be found in AnimatedContainer.js file

**Comment by jackw3434** [comment]
i jsut set swipeable to false. the pull down isnt necessary and can create my own close functionality but setting it to false means the user cant pull down the toast and cause a freeze

**Comment by sooonii** [comment]
> i jsut set swipeable to false. the pull down isnt necessary and can create my own close functionality but setting it to false means the user cant pull down the toast and cause a freeze

this works Thanks

**Comment by AF-appointhq** [comment]
> i jsut set swipeable to false. the pull down isnt necessary and can create my own close functionality but setting it to false means the user cant pull down the toast and cause a freeze

Thank you. It works. I tried with both "2.3.3" and "2.4.0" versions. and still happening. Your solution helped me. 👍🏻
I appreciate @calintamas. If you fixed this, please. Thank you.
